source/CMakeLists.txt

```diff
@@ -17,27 +17,35 @@
 #
 ###########################################################################
 
-# select correct source for target/toolchain
-set(UVISOR_PRESENT false)
-if     (TARGET_LIKE_FRDM_K64F)
-    if(TARGET_LIKE_GCC)
-        set(UVISOR_PRESENT true)
-        set(UVISOR_DIR "MK64FN1M0XXX12")        # FRDM-K64F
+# Determine the uVisor binary based on the target.
+# Note: We currently only support the GCC toolchain.
+set(UVISOR_SUPPORTED false)
+if(TARGET_LIKE_GCC)
+    # Pick the correct folder based on the target family.
+    if(TARGET_LIKE_KINETIS_K64_GCC)
+        set(UVISOR_FAMILY "kinetis")
+    elseif(TARGET_LIKE_STM32)
+        set(UVISOR_FAMILY "stm32")
```

Comment:
We should not throw an error. `UVISOR_SUPPORTED` is pessimistically set to `false` and updated to `true` if the target platform is either `kinetis` or `stm32` AND there exists a uVisor configuration for the specific yotta target.

So if `UVISOR_SUPPORTED` remain `false` uVisor is left out and a warning is thrown. That warning actually becomes an error if the target mistakenly stated that uVisor is present (`YOTTA_CFG_UVISOR_PRESENT`).